tasks/api-usage-analyze.js

```diff
@@ -0,0 +1,162 @@
+
+/**
+ * @typedef {Object} Options
+ * @property {Array<string>} input
+ * @property {'json'|'md'} format
+ * @property {boolean} withLegacyClient
+ */
+
+/**
+ * @param {string} option
+ * @param {number} i
+ */
+function getOptionValue(option, i) {
+  const next = i + 1;
+  if (next > process.argv.length - 1) {
+    throw new Error(`Invalid command: no value found for option "${option}"`);
+  }
+  return process.argv[next];
+}
+
+/**
+ * @returns {Options}
+ */
+function parseArgs() {
+  const input = [];
+  let format = 'json';
+  let withLegacyClient = false;
+  let apiVersion = null;
+
+  for (let i = 0; i < process.argv.length; i++) {
+    const arg = process.argv[i];
+    if (arg === '-i') {
+      input.push(getOptionValue(arg, i));
+      i++;
+    }
+    if (arg === '-f') {
+      format = getOptionValue(arg, i);
+      i++;
+    }
+    if (arg === '-l') {
+      withLegacyClient = true;
+    }
+    if (arg === '-v') {
+      apiVersion = getOptionValue(arg, i);
+      i++;
+    }
+
+    if (arg === '-h') {
+      const scriptName = path.basename(import.meta.url);
+      console.log(styleText('bold', 'Clever Cloud API calls analysis script'));
+      console.log();
+      console.log(styleText('italic', 'This tool scans JS code and detects all usage of Clever Cloud APIs.'));
+      console.log();
+      console.log(`node ${scriptName} -i DIR [-f FORMAT] [-l]`);
+      console.log();
+      console.log(styleText('bold', 'USAGE:'));
+      console.log();
+      console.log('-h               to display this help');
+      console.log('-i DIR           to include sources to scan (can be set multiple times)');
+      console.log('-f `json`|`md`   to define output format: json or markdown (`json` by default)');
+      console.log('-l               to collect only API calls done with legacy client');
+      console.log('-v               to filter by API calls version (`v2` or `v4)`');
+      console.log();
+      console.log(styleText('bold', 'EXAMPLES:'));
+      console.log();
+      console.log(`Scan multiple directories`);
+      console.log(`node ${scriptName} -i ~/dev/console3 -i ~/dev/clever-components -i ~/dev/clever-tools`);
+      console.log();
+      console.log(`Output in markdown format`);
+      console.log(`node ${scriptName} -i ~/dev/console3 -f md`);
+      console.log();
+      console.log(`Collect API calls done with legacy client only`);
+      console.log(`node ${scriptName} -i ~/dev/console3 -l`);
+      process.exit(0);
+    }
+  }
+
+  if (input.length === 0) {
+    throw new Error('Invalid command: no source input specified. Use "-i DIR" to include sources to scan.');
+  }
+  if (format !== 'json' && format !== 'md') {
+    throw new Error(`Invalid command: invalid format "${format}". Use "-f json" or "-f md".`);
+  }
+
+  if (apiVersion !== 'v2' && apiVersion !== 'v4') {
+    throw new Error(`Invalid command: invalid version "${apiVersion}". Use "-v v2" or "-v v4".`);
+  }
```

Comment:
question: are we supposed to accept `null`, this check is too strict

Reply:
Yes, nice catch that's my bad